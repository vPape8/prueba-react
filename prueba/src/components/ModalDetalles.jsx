// src/components/ModalDetalles.jsx
import React from 'react';
import '../assets/css/styleModal.css';

const ModalDetalles = ({ calculo, onClose }) => {
  if (!calculo) return null;

  const renderizarDetallesEspecificos = () => {
    if (!calculo.details) return <p>No hay detalles disponibles para este c치lculo.</p>;

    switch (calculo.tipo) {
      case 'comercial':
        return (
          <div className="detalles-especificos">
            <h4>游닍 Informaci칩n de Buque Comercial</h4>
            <div className="detalles-grid">
              <div className="detalle-item">
                <strong>Tonelaje:</strong>
                <span>{calculo.details.tonelaje} TRB</span>
              </div>
              <div className="detalle-item">
                <strong>D칤as de estad칤a:</strong>
                <span>{calculo.details.dias}</span>
              </div>
              <div className="detalle-item">
                <strong>Tipo de carga:</strong>
                <span>{calculo.details.tipo}</span>
              </div>
              <div className="detalle-item">
                <strong>Servicios:</strong>
                <span>{calculo.details.servicios}</span>
              </div>
            </div>
            
            <h4>游눯 Desglose de Costos</h4>
            <div className="costos-grid">
              <div className="costo-item">
                <strong>Tarifa base:</strong>
                <span>${calculo.details.tarifaBase?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Costos de servicios:</strong>
                <span>${calculo.details.costoServicios?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Impuestos:</strong>
                <span>${calculo.details.impuestos?.toFixed(2)}</span>
              </div>
            </div>
          </div>
        );
      
      case 'especial':
        return (
          <div className="detalles-especificos">
            <h4>游댧 Informaci칩n de Buque Especial</h4>
            <div className="detalles-grid">
              <div className="detalle-item">
                <strong>Tonelaje:</strong>
                <span>{calculo.details.tonelaje} TRB</span>
              </div>
              <div className="detalle-item">
                <strong>D칤as de estad칤a:</strong>
                <span>{calculo.details.dias}</span>
              </div>
              <div className="detalle-item">
                <strong>Tipo de buque:</strong>
                <span>{calculo.details.tipo}</span>
              </div>
              <div className="detalle-item">
                <strong>Servicios especiales:</strong>
                <span>{calculo.details.servicios}</span>
              </div>
            </div>
            
            <h4>游눯 Desglose de Costos</h4>
            <div className="costos-grid">
              <div className="costo-item">
                <strong>Tarifa base:</strong>
                <span>${calculo.details.tarifaBase?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Servicios especiales:</strong>
                <span>${calculo.details.especialServicios?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Impuestos:</strong>
                <span>${calculo.details.impuestos?.toFixed(2)}</span>
              </div>
            </div>
          </div>
        );
      
      case 'pasajero':
        return (
          <div className="detalles-especificos">
            <h4>游논 Informaci칩n de Buque de Pasaje</h4>
            <div className="detalles-grid">
              <div className="detalle-item">
                <strong>Eslora:</strong>
                <span>{calculo.details.eslora} metros</span>
              </div>
              <div className="detalle-item">
                <strong>N칰mero de pasajeros:</strong>
                <span>{calculo.details.pasajeros}</span>
              </div>
              <div className="detalle-item">
                <strong>D칤as de estad칤a:</strong>
                <span>{calculo.details.dias}</span>
              </div>
              <div className="detalle-item">
                <strong>Tipo de buque:</strong>
                <span>{calculo.details.tipo}</span>
              </div>
            </div>
            
            <h4>游눯 Desglose de Costos</h4>
            <div className="costos-grid">
              <div className="costo-item">
                <strong>Tarifa base:</strong>
                <span>${calculo.details.tarifaBase?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Tarifa por pasajero:</strong>
                <span>${calculo.details.tarifaPasajero?.toFixed(2)}</span>
              </div>
              <div className="costo-item">
                <strong>Impuestos:</strong>
                <span>${calculo.details.impuestos?.toFixed(2)}</span>
              </div>
            </div>
          </div>
        );
      
      default:
        return (
          <div className="detalles-especificos">
            <p>Detalles no disponibles para este tipo de c치lculo.</p>
            <pre style={{backgroundColor: '#f5f5f5', padding: '10px', borderRadius: '4px'}}>
              {JSON.stringify(calculo.details, null, 2)}
            </pre>
          </div>
        );
    }
  };

  return (
    <div 
      className="modal-overlay" 
      onClick={onClose}
      style={{
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        backgroundColor: 'rgba(0, 0, 0, 0.8)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        zIndex: 10000,
        padding: '20px'
      }}
    >
      <div 
        className="modal-content"
        onClick={(e) => e.stopPropagation()}
        style={{
          backgroundColor: 'white',
          padding: '30px',
          borderRadius: '12px',
          width: '90%',
          maxWidth: '700px',
          maxHeight: '85vh',
          overflow: 'auto',
          boxShadow: '0 20px 40px rgba(0, 0, 0, 0.3)',
          border: '1px solid #ddd'
        }}
      >
        {/* Header del Modal */}
        <div className="modal-header">
          <h2>游늵 Detalles del C치lculo</h2>
          <button className="modal-close" onClick={onClose}>칑</button>
        </div>

        {/* Informaci칩n General */}
        <div className="modal-general-info">
          <div className="info-grid">
            <div className="info-item">
              <strong>游댳 Tipo</strong>
              <span>{calculo.tipoDisplay}</span>
            </div>
            <div className="info-item">
              <strong>游늰 Fecha</strong>
              <span>{new Date(calculo.fecha).toLocaleString()}</span>
            </div>
            <div className="info-item">
              <strong>游눯 Total</strong>
              <span>${calculo.total.toFixed(2)}</span>
            </div>
          </div>
        </div>

        {/* Detalles Espec칤ficos */}
        {renderizarDetallesEspecificos()}

        {/* Footer del Modal */}
        <div className="modal-footer">
          <button className="btn-primary" onClick={onClose}>
            Cerrar
          </button>
        </div>
      </div>
    </div>
  );
};

export default ModalDetalles;