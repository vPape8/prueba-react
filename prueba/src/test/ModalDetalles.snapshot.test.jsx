import React from 'react';
import { render } from '@testing-library/react';
import ModalDetalles from '../components/ModalDetalles.jsx';
import '@testing-library/jest-dom';

test('el componente ModalDetalles coincide con el snapshot (testing-library)', () => {
  const calculoMock = {
    total: 1000,
    tipoCarga: 'general',
    details: {
      tarifaBase: 800,
      costoServicios: 150,
      impuestos: 50
    }
  };

  const { asFragment } = render(<ModalDetalles calculo={calculoMock} onClose={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});