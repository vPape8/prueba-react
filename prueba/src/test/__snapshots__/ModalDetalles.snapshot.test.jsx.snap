// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`el componente ModalDetalles coincide con el snapshot (testing-library) 1`] = `
<DocumentFragment>
  <div
    class="modal-overlay"
  >
    <div
      class="modal-content"
    >
      <h2>
        ðŸ“Š Detalles del CÃ¡lculo
      </h2>
      <button>
        Cerrar
      </button>
      <div>
        <strong>
          Tipo:
        </strong>
         
        <span>
          general
        </span>
      </div>
      <div>
        <strong>
          Total:
        </strong>
         
        <span>
          $1000.00
        </span>
      </div>
      <div>
        <p>
          Tarifa base: $800.00
        </p>
        <p>
          Servicios: $150.00
        </p>
        <p>
          Impuestos: $50.00
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
